rust/core-lib/src/tabs.rs

```diff
@@ -453,3 +459,67 @@ impl<W: Write> DocumentCtx<W> {
         self.update_channel.send((view_id, update, undo_group)).unwrap();
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use xi_rpc::{RpcLoop};
+    use std::env;
+    use std::fs::File;
+
+    // a bit of gymnastics to let us instantiate an Editor instance
+    fn mock_doc_ctx(tempfile: &str) -> DocumentCtx<File> {
+        let mut dir = env::temp_dir();
+        dir.push(tempfile);
+        let f = File::create(dir).unwrap();
+
+        let mock_loop = RpcLoop::new(f);
+        let mock_peer = mock_loop.get_peer();
+        let (update_tx, _) = mpsc::channel();
+
+        DocumentCtx {
+            kill_ring: Arc::new(Mutex::new(Rope::from(""))),
+            rpc_peer: mock_peer.clone(),
+            style_map: Arc::new(Mutex::new(StyleMap::new())),
+            update_channel: update_tx,
+        }
+    }
+
+    #[test]
+    fn test_save_as() {
+        let container_ref = BufferContainerRef::new();
+        assert!(!container_ref.has_open_file("a fake file, for sure"));
+        let view_id_1 = "view-id-1".to_owned();
+        let buf_id_1 = "buf-id-1".to_owned();
+        let path_1 = PathBuf::from("a_path");
+        let path_2 = PathBuf::from("a_different_path");
+        let editor = Editor::new(mock_doc_ctx(&view_id_1), &view_id_1);
+        container_ref.add_editor(&view_id_1, &buf_id_1, editor, None);
+        assert_eq!(container_ref.lock().editors.len(), 1);
+
+        // save this somewhere:
```

Comment:
This says "save" but I don't see any file writing, just setting paths. Am I missing something?